File: Java/2515. Shortest Distance to Target String in a Circular Array.java
class Solution {
    public int closestTarget(String[] words, String target, int startIndex) {
        int N = words.length;

        for (int i = 0; i < N; i++) {
            int l = (startIndex - i + N) % N;
            int r = (startIndex + i) % N;

            if (target.equals(words[l]) || target.equals(words[r])) {
                return i;
            }
        }

        return -1;
    }
}

File: Python3/2515. Shortest Distance to Target String in a Circular Array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        N = len(words)

        for i in range(N):
            l = (startIndex - i + N) % N
            r = (startIndex + i) % N

            if words[l] == target or words[r] == target:
                return i

        return -1